CounterApp/Lock/LockCore.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  LockCore.swift
+//  CounterApp
+//
+//  Created by Wenjuan Li on 2022/7/12.
+//
+
+import ComposableArchitecture
+import Foundation
+
+struct LockState: Equatable {
+    var counters: IdentifiedArrayOf<CounterState> = []
+    var isPresent: Bool = false
+    var unlockAlert: AlertState<LockAction>?
+}
+
+enum LockAction: Equatable {
+    case counter(id: CounterState.ID, action: CounterAction)
+    case setSheet(isPresented: Bool)
+    case alertDismissed
+}
+
+struct LockEnvironment {
+    var tryToUnlock: (IdentifiedArrayOf<CounterState>) -> Bool
+
+    static func defaultEnv() -> LockEnvironment {
+        LockEnvironment(tryToUnlock: { counters in
+            counters.count == 3
+                && counters[0].count == 9
+                && counters[1].count == 5
+                && counters[2].count == 7
+        })
+    }
```

Comment:
Nice! As a further exercise, could you refactor this so this logic lives in `lockReducer`?

Reply:
@nanocba I just move the logic to `lockReducer`, is this what you mean?